src/Prophecy.phpstub

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Prophecy\Prophecy {
+    /** @template T */
+    class ObjectProphecy implements ProphecyInterface
+    {
+        /** @return T */
+        public function reveal() {}
+    }
+
+    /**
+     * @param string $methodName
+     * @param mixed[] $arguments
+     * @return MethodProphecy
+     */
+    public function __call($methodName, array $arguments): MethodProphecy {}
```

Comment:
This is not enough: it declares that after the first call you get a `MethodProphecy`, but we need a way to say that `$methodName` must match a method that exists on the original object.

Reply:
I'd personally leave the MethodsClassReflectionExtension in for this.